numpy/core/src/umath/ufunc_object.c

```diff
@@ -1867,6 +1880,34 @@ _has_output_coredims(PyUFuncObject *ufunc) {
     return 0;
 }
 
+/*
+ * Check whether the gufunc can be used with keepdims, i.e., that all its
+ * input arguments have the same number of core dimension, and all output
+ * arguments have no core dimensions. Sets an error if this is not the case.
+ */
+static int
+_check_keepdims_support(PyUFuncObject *ufunc) {
```

Comment:
This function should return an error code (0 / -1) not a boolean (1, 0), since it's called `_check_support` not `_is_support` (and modifies state)